### PR TITLE
Allow column renames of fuse input

### LINKED ISSUE #1090: Coalesce transformer cannot be opened if the units are not updateable
If a coalesce transformer is configured with a column that has been renamed, then the unit is in an inconsistent state. This is unfortunate in itself, but a bigger problem for the user is that he cannot even open the dialog to change the way it is set up.

Here is the stack trace that an error popup will show:
```

Unexpected error!
 

Message:
    java.lang.IllegalStateException: Column 'JavaScript output WW' not found. Available columns: [CUSTOMERNUMBER, CUSTOMERNAME, CONTACTLASTNAME, CONTACTFIRSTNAME, PHONE, ADDRESSLINE1, ADDRESSLINE2, CITY, STATE, POSTALCODE, COUNTRY, SALESREPEMPLOYEENUMBER, CREDITLIMIT, JOBTITLE, COUNTRY (standardized), JavaScript output W, JavaScript output UK/US]
Level:
    SEVERE
Stack Trace:
Column 'JavaScript output WW' not found. Available columns: [CUSTOMERNUMBER, CUSTOMERNAME, CONTACTLASTNAME, CONTACTFIRSTNAME, PHONE, ADDRESSLINE1, ADDRESSLINE2, CITY, STATE, POSTALCODE, COUNTRY, SALESREPEMPLOYEENUMBER, CREDITLIMIT, JOBTITLE, COUNTRY (standardized), JavaScript output W, JavaScript output UK/US]
    org.datacleaner.components.fuse.CoalesceUnit.updateInputColumns(CoalesceUnit.java:145)
    org.datacleaner.panels.fuse.CoalesceUnitPanel.<init>(CoalesceUnitPanel.java:98)
    org.datacleaner.panels.fuse.ColumnListMultipleCoalesceUnitPropertyWidget.addCoalesceUnit(ColumnListMultipleCoalesceUnitPropertyWidget.java:152)
    org.datacleaner.panels.fuse.ColumnListMultipleCoalesceUnitPropertyWidget$3.run(ColumnListMultipleCoalesceUnitPropertyWidget.java:126)
    org.datacleaner.widgets.properties.AbstractPropertyWidget.batchUpdateWidget(AbstractPropertyWidget.java:87)
    org.datacleaner.panels.fuse.ColumnListMultipleCoalesceUnitPropertyWidget.<init>(ColumnListMultipleCoalesceUnitPropertyWidget.java:122)
    org.datacleaner.panels.fuse.CoalesceMultipleFieldsTransformerComponentBuilderPresenter.<init>(CoalesceMultipleFieldsTransformerBuilderPresenter.java:54)
    org.datacleaner.panels.fuse.CoalesceMultipleFieldsTransformerComponentBuilderPresenterRenderer.render(CoalesceMultipleFieldsTransformerComponentBuilderPresenterRenderer.java:67)
    org.datacleaner.panels.fuse.CoalesceMultipleFieldsTransformerComponentBuilderPresenterRenderer.render(CoalesceMultipleFieldsTransformerComponentBuilderPresenterRenderer.java:1)
    org.datacleaner.windows.ComponentConfigurationDialog.getDialogContent(ComponentConfigurationDialog.java:176)
    org.datacleaner.windows.AbstractDialog.getWindowContent(AbstractDialog.java:174)
    org.datacleaner.windows.AbstractDialog.initialize(AbstractDialog.java:124)
    org.datacleaner.windows.AbstractDialog.setVisible(AbstractDialog.java:147)
    org.datacleaner.windows.AbstractDialog.open(AbstractDialog.java:97)
    org.datacleaner.widgets.visualization.JobGraphActions.showConfigurationDialog(JobGraphActions.java:81)
    org.datacleaner.widgets.visualization.JobGraphMouseListener.onComponentDoubleClicked(JobGraphMouseListener.java:107)
    org.datacleaner.widgets.visualization.JobGraphMouseListener.graphPressed(JobGraphMouseListener.java:330)
    edu.uci.ics.jung.visualization.control.MouseListenerTranslator.mousePressed(MouseListenerTranslator.java:77)
    java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:280)
    java.awt.AWTEventMulticaster.mousePressed(AWTEventMulticaster.java:279)
```

To reproduce, you can use this analysis XML job:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<job xmlns="http://eobjects.org/analyzerbeans/job/1.0">
    <job-metadata>
        <metadata-properties>
            <property name="CoordinatesY.PUBLIC.EMPLOYEES">40</property>
            <property name="CoordinatesX.PUBLIC.CUSTOMERS">68</property>
            <property name="CoordinatesY.PUBLIC.CUSTOMERS">146</property>
            <property name="CoordinatesX.PUBLIC.EMPLOYEES">40</property>
        </metadata-properties>
    </job-metadata>
    <source>
        <data-context ref="orderdb"/>
        <columns>
            <column id="col_customernumber" path="CUSTOMERS.CUSTOMERNUMBER" type="INTEGER"/>
            <column id="col_customername" path="CUSTOMERS.CUSTOMERNAME" type="VARCHAR"/>
            <column id="col_contactlastname" path="CUSTOMERS.CONTACTLASTNAME" type="VARCHAR"/>
            <column id="col_contactfirstname" path="CUSTOMERS.CONTACTFIRSTNAME" type="VARCHAR"/>
            <column id="col_phone" path="CUSTOMERS.PHONE" type="VARCHAR"/>
            <column id="col_addressline1" path="CUSTOMERS.ADDRESSLINE1" type="VARCHAR"/>
            <column id="col_addressline2" path="CUSTOMERS.ADDRESSLINE2" type="VARCHAR"/>
            <column id="col_city" path="CUSTOMERS.CITY" type="VARCHAR"/>
            <column id="col_state" path="CUSTOMERS.STATE" type="VARCHAR"/>
            <column id="col_postalcode" path="CUSTOMERS.POSTALCODE" type="VARCHAR"/>
            <column id="col_country" path="CUSTOMERS.COUNTRY" type="VARCHAR"/>
            <column id="col_salesrepemployeenumber" path="CUSTOMERS.SALESREPEMPLOYEENUMBER" type="INTEGER"/>
            <column id="col_creditlimit" path="CUSTOMERS.CREDITLIMIT" type="NUMERIC"/>
            <column id="col_jobtitle" path="CUSTOMERS.JOBTITLE" type="VARCHAR"/>
        </columns>
    </source>
    <transformation>
        <transformer>
            <descriptor ref="Country standardizer"/>
            <metadata-properties>
                <property name="CoordinatesY">141</property>
                <property name="CoordinatesX">299</property>
            </metadata-properties>
            <properties>
                <property name="Default country" value="&lt;null&gt;"/>
                <property name="Output format" value="ISO2"/>
            </properties>
            <input ref="col_country"/>
            <output id="col_countrystandardized" name="COUNTRY (standardized)"/>
        </transformer>
        <transformer requires="outcome_0" name="Rest of the world JS">
            <descriptor ref="JavaScript transformer (simple)"/>
            <metadata-properties>
                <property name="CoordinatesY">263</property>
                <property name="CoordinatesX">758</property>
            </metadata-properties>
            <properties>
                <property name="Return type" value="STRING"/>
                <property name="Source code">function eval() {
	return &amp;quot;WW &amp;quot; + values&amp;#91;0&amp;#93;;
}

eval();</property>
            </properties>
            <input ref="col_state"/>
            <output id="col_javascriptoutputww" name="JavaScript output WW"/>
        </transformer>
        <transformer requires="outcome_1" name="US/UK javascript">
            <descriptor ref="JavaScript transformer (simple)"/>
            <metadata-properties>
                <property name="CoordinatesY">361</property>
                <property name="CoordinatesX">452</property>
            </metadata-properties>
            <properties>
                <property name="Return type" value="STRING"/>
                <property name="Source code">function eval() {
	return &amp;quot;UK/US &amp;quot; + values&amp;#91;0&amp;#93;;
}

eval();</property>
            </properties>
            <input ref="col_state"/>
            <output id="col_javascriptoutputukus" name="JavaScript output UK/US"/>
        </transformer>
        <transformer>
            <descriptor ref="Fuse / Coalesce fields"/>
            <metadata-properties>
                <property name="CoordinatesY">462</property>
                <property name="CoordinatesX">695</property>
            </metadata-properties>
            <properties>
                <property name="Consider empty string as null" value="true"/>
                <property name="Units" value="[&amp;#91;JavaScript output WW&amp;#44;JavaScript output UK/US&amp;#93;]"/>
            </properties>
            <input ref="col_javascriptoutputww"/>
            <input ref="col_javascriptoutputukus"/>
            <output id="col_fusecoalescefields1" name="Fuse / Coalesce fields (1)"/>
        </transformer>
        <filter>
            <descriptor ref="Equals"/>
            <metadata-properties>
                <property name="CoordinatesY">113</property>
                <property name="CoordinatesX">486</property>
            </metadata-properties>
            <properties>
                <property name="Compare values" value="[GB,US]"/>
            </properties>
            <input ref="col_countrystandardized" name="Input column"/>
            <outcome id="outcome_1" category="EQUALS"/>
            <outcome id="outcome_0" category="NOT_EQUALS"/>
        </filter>
    </transformation>
    <analysis>
        <analyzer>
            <descriptor ref="Mark rows"/>
            <metadata-properties>
                <property name="CoordinatesY">439</property>
                <property name="CoordinatesX">869</property>
            </metadata-properties>
            <properties>
                <property name="Condition description" value="..."/>
            </properties>
            <input ref="col_fusecoalescefields1"/>
            <input ref="col_customernumber"/>
            <input ref="col_country"/>
        </analyzer>
    </analysis>
</job>
```

Try renaming one of the output columns of the JS transformers. Then this error happens when double clicking on the ```Fuse / Coalesce fields``` node.

### PR DESCRIPTION
**Pull request by LosD**
This makes the fuse/coalesce transformer survive column renames. The connections will still be lost, though.

Fixes #1090

### CODE REVIEW

File: desktop/api/src/main/java/org/datacleaner/panels/TransformerComponentBuilderPanel.java [hunk 1]
```diff
@@ -85,10 +88,11 @@ protected TransformerComponentBuilderPanel(Image watermarkImage, int watermarkHo
 
         final List<MutableInputColumn<?>> outputColumns;
         if (_componentBuilder.isConfigured()) {
-            outputColumns = _componentBuilder.getOutputColumns();
+            outputColumns = safeGetOutputColumns(transformerJobBuilder);
         } else {
-            outputColumns = new ArrayList<MutableInputColumn<?>>(0);
+            outputColumns = new ArrayList<>(0);
```
Comment (kaspersorensen) [inline desktop/api/src/main/java/org/datacleaner/panels/TransformerComponentBuilderPanel.java]: Slight improvement would be to use ```Collections.emptyList()```, but not a blocker.

File: desktop/api/src/main/java/org/datacleaner/panels/TransformerComponentBuilderPanel.java [hunk 2]
```diff
@@ -138,6 +142,16 @@ public void actionPerformed(ActionEvent e) {
         });
     }
 
+    private List<MutableInputColumn<?>> safeGetOutputColumns(
+            final TransformerComponentBuilder<?> transformerJobBuilder) {
+        try {
+            return _componentBuilder.getOutputColumns();
+        } catch (Exception e) {
+            logger.warn("Could not get outputColumns for transformer {}", transformerJobBuilder.getName(), e);
+            return new ArrayList<>(0);
```
Comment (kaspersorensen) [inline desktop/api/src/main/java/org/datacleaner/panels/TransformerComponentBuilderPanel.java]: Same here.

**Comment by kaspersorensen** [comment]
LGTM. Feel free to merge at will (unless you want the empty list in which case you could do that and then merge after Travis agrees).

File: desktop/api/src/main/java/org/datacleaner/panels/TransformerComponentBuilderPanel.java (hunk 1 above)
Comment (LosD) [inline desktop/api/src/main/java/org/datacleaner/panels/TransformerComponentBuilderPanel.java]: Agreed, it's a few seconds change, so let's do it

**Comment by LosD** [comment]
Done, let's see if Travis likes it.

**Pull request merged**